### PR TITLE
Checks 'None' after using match

### PR DESCRIPTION
**Pull request by vishalg2235**
Fixes https://github.com/sympy/sympy/issues/14624 .
Checked `m` for None value.
Output of this branch(It took very long time for this unsolved output): 
```
>>> sym.inverse_fourier_transform(a, w, -t)
R*InverseFourierTransform(1/(-lambda + I*omega), omega, -t) + InverseFourierTransform(1/(I*omega - conjugate(lambda)), omega, -t)*conjugate(R)
```
@asmeurer please review.

### CODE REVIEW

File: sympy/integrals/heurisch.py [hunk 1]
```diff
@@ -611,10 +611,11 @@ def _integrate(field=None):
         x, y = symbols('x y', cls=Wild, exclude=[I])
         for poly in list(irreducibles):
             m = poly.match(x + I*y)
```
Comment (smichr) [inline sympy/integrals/heurisch.py]: or using @jksuom suggestion, maybe something like this:
```
ip = collect(poly, I)
if ip:
    rp = poly.subs(I, 0)
    if rp != poly - expand_mul(I*ip):
        raise NotImplementedError
    pairs.add(rp, ip)
    irreducibles.remove(poly)
```

Comment (jksuom) [reply sympy/integrals/heurisch.py]: It seems that `collect(poly, I, evaluate=False)` should be used to separate terms involving `I`.

Comment (smichr) [reply sympy/integrals/heurisch.py]: So thus?
```
m = collect(poly, I, evaluate=False)
y = m.get(I, 0)
if y:
    x = m.get(1, 0)
    if x + I*y != poly:
        raise NotImplementedError
    pairs.add((x, y))
    irreducibles.remove(poly)
```

Comment (jksuom) [reply sympy/integrals/heurisch.py]: I'm not sure if the numbers should be SymPy objects (like `m.get(S.One, S.Zero)`). Maybe `continue` should suffice if `x + I*y != poly` (is that even possible?). But else this looks promising.

Comment (smichr) [reply sympy/integrals/heurisch.py]: modified

**Pull request merged**